mycroft/skills/skill_updater.py

```diff
@@ -0,0 +1,251 @@
+        """Determine the initial values of the next/last download times.
+
+        Update immediately if the .msm or installed skills file is missing
+        otherwise use the timestamp on .msm as a basis.
+        """
+        msm_files_exist = (
+                os.path.exists(self.dot_msm_path) and
+                os.path.exists(self.installed_skills_file_path)
+        )
+        if msm_files_exist:
+            mtime = os.path.getmtime(self.dot_msm_path)
+            next_download = mtime + self.update_interval
+        else:
+            # Last update can't be found or the requirements don't seem to be
+            # installed trigger update before skill loading
+            next_download = time() - 1
+
+        return next_download
+
+    @property
+    def config(self):
+        """Property representing the device configuration."""
+        return Configuration.get()
+
+    @property
+    def installed_skills_file_path(self):
+        """Property representing the path of the installed skills file."""
+        if self._installed_skills_file_path is None:
+            virtual_env_path = os.path.dirname(os.path.dirname(sys.executable))
+            if os.access(virtual_env_path, os.W_OK | os.R_OK | os.X_OK):
+                self._installed_skills_file_path = os.path.join(
+                    virtual_env_path,
+                    '.mycroft-skills'
+                )
+            else:
+                self._installed_skills_file_path = os.path.expanduser(
+                    '~/.mycroft/.mycroft-skills'
+                )
+
+        return self._installed_skills_file_path
+
+    @property
+    def default_skill_names(self) -> tuple:
+        """Property representing the default skills expected to be installed"""
+        default_skill_groups = dict(self.msm.repo.get_default_skill_names())
+        default_skills = set(default_skill_groups['default'])
+        platform_default_skills = default_skill_groups.get(self.msm.platform)
+        if platform_default_skills is None:
+            log_msg = 'No default skills found for platform {}'
+            LOG.info(log_msg.format(self.msm.platform))
+        else:
+            default_skills.update(platform_default_skills)
+
+        return tuple(default_skills)
+
+    def _load_installed_skills(self):
+        """Load the last known skill listing from a file."""
+        if os.path.isfile(self.installed_skills_file_path):
+            with open(self.installed_skills_file_path) as skills_file:
+                self.installed_skills = {
+                    i.strip() for i in skills_file.readlines() if i.strip()
+                }
+
+    def _save_installed_skills(self):
+        """Save the skill listing after the download to a file."""
+        with open(self.installed_skills_file_path, 'w') as skills_file:
+            for skill_name in self.installed_skills:
+                skills_file.write(skill_name + '\n')
+
+    def update_skills(self, speak=False, quick=False):
```

Comment:
I am glad you mentioned that.  I couldn't find where the speak parameter was used but was afraid to get rid of it in case there was some sort of backwards compatibility issue.  Happy to remove dead code!